src/flint/FlintTypes.jl

```diff
@@ -3001,22 +3001,23 @@ A $p$-adic field for some prime $p$.
    mode::Cint
    prec_max::Int
 
-   function PadicField(p::ZZRingElem, prec::Int; cached::Bool = true, check::Bool = true)
-      check && !is_probable_prime(p) && throw(DomainError(p, "Characteristic must be prime"))
+   function PadicField(p::ZZRingElem, prec::Int = 64; cached::Bool = true, check::Bool = true)
+      check && !is_probable_prime(p) && throw(DomainError(p, "Integer must be prime"))
 
-      return get_cached!(PadicBase, (p, prec), cached) do
+      Qp = get_cached!(PadicBase, p, cached) do
          d = new()
          ccall((:padic_ctx_init, libflint), Nothing,
                (Ref{PadicField}, Ref{ZZRingElem}, Int, Int, Cint),
                d, p, 0, 0, 0)
          finalizer(_padic_ctx_clear_fn, d)
-         d.prec_max = prec
          return d
       end
+      Qp.prec_max = prec
```

Comment:
stupid question maybe, but shouldn't this be something like `Qp.prec_max = max(prev, Qp.prec_mac)`?

Reply:
The question when and where we store the precision is still open for debate. I don't use local fields (or other inexact rings), so I leave that for others to decide.